realm/realm-library/src/main/cpp/io_realm_mongodb_sync_Sync.cpp

```diff
@@ -73,3 +74,34 @@ JNIEXPORT void JNICALL Java_io_realm_mongodb_sync_Sync_nativeCreateSession(JNIEn
     }
     CATCH_STD()
 }
+
+JNIEXPORT jstring JNICALL Java_io_realm_mongodb_sync_Sync_nativeGetPathForRealm(JNIEnv* env,
+                                                                                jclass,
+                                                                                jstring j_user_id,
+                                                                                jstring j_encoded_partition_value,
+                                                                                jstring j_override_filename)
+{
+    try {
+        // This is a little bit of a hack. Normally Realm Java doesn't generate the C++ SyncConfig
+        // until the Realm is opened, but the Sync API for creating the Realm path require that
+        // it is created up front. So we cheat and create a SyncConfig with the minimal values
+        // needed for the path to be calculated.
+        JStringAccessor user_id(env, j_user_id);
+        std::shared_ptr<SyncUser> user = SyncManager::shared().get_existing_logged_in_user(user_id);
```

Comment:
The primary reason is that asking for a potentially logged out user requires both the access/refresh token as well, so it would complicate this code for a very exotic use case (which would also be a wrong use of the API).

We already check if the user is logged in in the SyncConfiguration constructor, so this should be quite safe unless people do this:

```
val builder = SyncConfiguration.Builder(user, "partition")
user.logOut()
val config = builder.build() // Will throw with the current code
```